### ISSUE TITLE
MacOS Print Issues

### ISSUE THREAD

**Issue by mrdanbrooks** [typora/typora-issues#5865]
- [x] Searched existing issues to avoid creating duplicates.

- [x] Confirmed that it can be reproduced in built-in themes without customized css.

- [x] Searched http://support.typora.io/

----

**Describe the bug**
The print dialog in MacOS fails to show the correct preview of the document, and fails to print more than one page (either to a printer or using print to pdf).

**To Reproduce**
- When selecting File > Print on a long (multipage) document, only the first page has any content in the preview section. (see image below)
- If I then click "Print", only page 1 prints. No blank pages are printed, only a single, one sided page that looks like "Page 1" in the preview gets printed.
- If from the Print dialog, I select "PDF" > "Open in Preview", I again see only the first page, and no blank pages.
- If from the Print dialog, I select "PDF" > "Save as PDF", the resulting file only has the first page, and no blank pages.
- Inside the Print dialog, double-sided print is "off" and unable to be changed, even for a duplex printer that Preview.app allows you to set the same option for.

**Expected behavior**
- The print preview should show the correct number of pages, populated with text.
- Printing to a printer should produce more than just the first page for multipage documents.
- Duplex printing should be changeable for duplex printers.
- Printing to pdf should produce a multipage document with all content, similar to using the export to pdf functionality.

**Workarounds**
- If I go to "Export" > "PDF", the resulting file has 2 pages that appears to contain all of the contents of the document (all the way to item 64). Printing the pdf from Preview.app allows me to use my duplex printer to print double-sided.
- From the "File" > "Print" dialog, if I change the orientation from "Portrait", to "Landscape", and then back to "Portrait", the preview correctly updates to show 2 pages worth of text. At that point, I can either print the 2 page document or save a 2 page pdf document. The double-sided option remains grayed out and in "off" mode, so I when using a printer I can only print singled sided directly from the app.

**Screenshots / Screencasts**
<img width="774" alt="Screenshot 2023-10-10 at 12 16 59 PM" src="https://github.com/typora/typora-issues/assets/5226494/36d8127a-5787-4d08-b056-67b358e84afc">

**Sample Markdown File**
[Test.md](https://github.com/typora/typora-issues/files/12859703/Test.md)

**Desktop (please complete the following information):**
 - MacOS Ventura 13.2.1

**Typora Version**
- Typora 1.7.5

**Additional context**
I have seen multiple other issues related to printing, print preview, and pdf generation. While I suspect some of these may be related, they all appear to be reported from Linux or Windows, and do not appear to be this particular issue of only printing the first page.
- [Cannot Print on Typora #5033](https://github.com/typora/typora-issues/issues/5033)
- [ERR:Failed to export as PDF. undefined #5785](https://github.com/typora/typora-issues/issues/5785)
- [Cannot Export PDF #5037](https://github.com/typora/typora-issues/issues/5037)
- [Export / Print Preview #1732](https://github.com/typora/typora-issues/issues/1732)

**Comment by abnerlee** [comment]
Does this still occur?

We cannot reproduce this on macOS 14 and Typora 1.10

**Comment by mrdanbrooks** [comment]
Yes, this is still occurring for me on MacOS 13.2.1 with Typora 1.8.10.
Same symptoms.

Dan

**Comment by oskapt** [comment]
This is happening with me too. MacOS 13.6.6 with Typora 1.8.10. A workaround I discovered today is to go into typewriter mode and then print from there. It then switches back to the view mode and shows all pages. Video attached.

https://github.com/typora/typora-issues/assets/171292/a09a80f6-d53d-4a33-96d3-8231e6bb2a80

**Comment by carpecrustum** [comment]
Yes, this is happening for me too.Typora version 1.9.4 on Mac OS 13.0

I'm shown four pages in the print dialog sidebar but only one prints. If I try to select the pages to print using the "Range from" option and specifying pages 1 through 4, I'm told the range is invalid.

**Comment by rmschne** [comment]
Seeing this with macOS 13.6.7 and Typora 1.9.4.  One page only of a 2 page md document.  The typewriter workaround mentioned above "oskapt" does not work here.

**Comment by abnerlee** [comment]
may relates to macOS 13, since no reports from macOS 14

**Comment by oskapt** [comment]
No longer seeing the issue with macOS 15.3.
